name: events-to-datadog-bridge

on: [pull_request]

jobs:
  node-16:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2

      - uses: actions/setup-node@v2
        with:
          node-version: "16.x"

      - uses: actions/cache@v2
        id: npm-cache
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-

      - name: npm install
        run: |
          npm ci

      - run: npm run lint

      - run: npm run test:cov
        env:
          DATADOG_SERVICE_NAME: test-events-ci
          DATADOG_URL: app.datadoghq.eu
          DATADOG_API_KEY: test-key
          EVENT_NAME: test-event

      - run: npm run test:e2e -- --coverage
        env:
          DATADOG_SERVICE_NAME: test-events-ci
          DATADOG_URL: app.datadoghq.eu
          DATADOG_API_KEY: test-key
          EVENT_NAME: test-event

      - name: Codecov
        uses: codecov/codecov-action@v2
        with:
          fail_ci_if_error: true
          token: ${{ secrets.CODECOV_TOKEN }}
          file: coverage/lcov.info
          flags: unit
